api/e2e/test/config/config.go

```diff
@@ -121,6 +153,21 @@ func LoadFromFiles(filepaths ...string) (*Config, error) {
 		return nil, fmt.Errorf("failed to unmarshal config values: %s", err)
 	}
 
+	// NOTE: This section is added to parse any fields in k8sConfig that does not
+	// have yaml tags.
+	// For example `certificate-authority-data` is not unmarshalled
+	// by vipers unmarshal method.
```

Comment:
We actually borrowed the cluster and user fields to use in the `k8sConfig` struct. Cluster & user correspond to the [Cluster](https://pkg.go.dev/k8s.io/client-go/tools/clientcmd/api/v1#Cluster) struct and [AuthInfo](https://pkg.go.dev/k8s.io/client-go/tools/clientcmd/api/v1#AuthInfo) struct defined in client-go clientcmd/api/v1 package respectively. 

In the struct the `certifiacte-authority-data` field (and a few other fieds) only have json tags defined because kubernetes uses their own [yaml package](https://pkg.go.dev/sigs.k8s.io/yaml@v1.3.0) to do the unmarshalling/marshalling. 

This is a bit hacky but at least it is able to read the config.

Reply:
Ohhh I see, oops my bad I should've looked into the `Cluster` and `AuthInfo` structs that you've quoted in the PR description 😅 Yep I don't have anything against your implementation of the unmarshalling/marshalling! Thanks for devising this way to parse the configs!